extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/db/mysql/MySQLBasicSecurityStorageConnector.java

```diff
@@ -0,0 +1,175 @@
+
+  @Inject
+  public MySQLBasicSecurityStorageConnector(
+      Supplier<MetadataStorageConnectorConfig> config,
+      Supplier<BasicAuthConfig> basicAuthConfigSupplier,
+      ObjectMapper jsonMapper
+  )
+  {
+    super(config, basicAuthConfigSupplier, jsonMapper);
+
+    final BasicDataSource datasource = getDatasource();
+    datasource.setDriverClassLoader(getClass().getClassLoader());
+    datasource.setDriverClassName("com.mysql.jdbc.Driver");
+
+    // use double-quotes for quoting columns, so we can write SQL that works with most databases
+    datasource.setConnectionInitSqls(ImmutableList.of("SET sql_mode='ANSI_QUOTES'"));
+
+    this.dbi = new DBI(datasource);
+    log.info("Configured MySQL as security storage");
+  }
+
+  @Override
+  public void createRoleTable()
+  {
+    createTable(
+        ROLES,
+        ImmutableList.of(
+            StringUtils.format(
+                "CREATE TABLE %1$s (\n"
+                + "  name VARCHAR(255) NOT NULL,\n"
+                + "  PRIMARY KEY (name),\n"
+                + "  UNIQUE (name)\n"
+                + ")",
+                ROLES
+            )
+        )
+    );
+  }
+
+  @Override
+  public void createUserTable()
+  {
+    createTable(
+        USERS,
+        ImmutableList.of(
+            StringUtils.format(
+                "CREATE TABLE %1$s (\n"
+                + "  name VARCHAR(255) NOT NULL,\n"
+                + "  PRIMARY KEY (name),\n"
+                + "  UNIQUE (name)\n"
+                + ")",
+                USERS
+            )
+        )
+    );
+  }
+
+  @Override
+  public void createPermissionTable()
+  {
+    createTable(
+        PERMISSIONS,
+        ImmutableList.of(
+            StringUtils.format(
+                "CREATE TABLE %1$s (\n"
+                + "  id INTEGER NOT NULL AUTO_INCREMENT,\n"
+                + "  resource_json BLOB(1024) NOT NULL,\n"
+                + "  role_name VARCHAR(255) NOT NULL, \n"
+                + "  PRIMARY KEY (id),\n"
+                + "  FOREIGN KEY (role_name) REFERENCES roles(name) ON DELETE CASCADE\n"
+                + ")",
+                PERMISSIONS
+            )
+        )
+    );
+  }
+
+  @Override
+  public void createUserCredentialsTable()
+  {
+    createTable(
+        USER_CREDENTIALS,
+        ImmutableList.of(
+            StringUtils.format(
+                "CREATE TABLE %1$s (\n"
+                + "  user_name VARCHAR(255) NOT NULL, \n"
+                + "  salt BLOB(32) NOT NULL, \n"
+                + "  hash BLOB(64) NOT NULL, \n"
+                + "  iterations INTEGER NOT NULL, \n"
+                + "  PRIMARY KEY (user_name), \n"
+                + "  FOREIGN KEY (user_name) REFERENCES users(name) ON DELETE CASCADE\n"
+                + ")",
+                USER_CREDENTIALS
+            )
+        )
+    );
+  }
+
+
+  @Override
+  public boolean tableExists(Handle handle, String tableName)
+  {
+    // ensure database defaults to utf8, otherwise bail
```

Comment:
bail -> fail?

Reply:
"bail" is used here in the sense of "abandon what it's doing"